contracts/modules/capital/SwapOperator.sol

```diff
@@ -295,6 +314,81 @@ contract SwapOperator is ReentrancyGuard {
     emit Swapped(ETH, stETH, amountIn, amountOut);
   }
 
+  function swapETHForEnzymeVaultShare(uint amountIn, uint amountOutMin) external onlySwapController {
+    IPool pool = _pool();
+    IEnzymeV4Comptroller comptrollerProxy = IEnzymeV4Comptroller(IEnzymeV4Vault(enzymeV4VaultProxyAddress).getAccessor());
+    address toTokenAddress = enzymeV4VaultProxyAddress;
+    address weth = router.WETH();
+
+    (
+    uint112 minAmount,
+    uint112 maxAmount,
+    /* uint32 lastAssetSwapTime */,
+    /* uint maxSlippageRatio */
+    ) = pool.getAssetDetails(toTokenAddress);
+
+    require(!(minAmount == 0 && maxAmount == 0), "SwapOperator: asset is not enabled");
+
+    uint balanceBefore = IERC20(toTokenAddress).balanceOf(address(pool));
+
+    pool.transferAssetToSwapOperator(ETH, amountIn);
+
+    require(comptrollerProxy.getDenominationAsset() == weth, "SwapOperator: invalid denomination asset");
+
+    enzymeV4DepositWrapper.exchangeEthAndBuyShares{value: amountIn}(address(comptrollerProxy), weth, 1, address(0), address(0), '0x', 0);
+
+    pool.setAssetDataLastSwapTime(toTokenAddress, uint32(block.timestamp));
+
+    uint amountOut = IERC20(toTokenAddress).balanceOf(address(this));
+
+    require(amountOut >= amountOutMin, "SwapOperator: amountOut < amountOutMin");
+    require(balanceBefore < minAmount, "SwapOperator: balanceBefore >= min");
+    require(balanceBefore + amountOutMin <= maxAmount, "SwapOperator: balanceAfter > max");
+
+    {
+      uint ethBalanceAfter = address(pool).balance;
+      require(ethBalanceAfter >= pool.minPoolEth(), "SwapOperator: insufficient ether left");
+    }
+
+    transferAssetTo(enzymeV4VaultProxyAddress, address(pool), amountOut);
+
+    emit Swapped(ETH, enzymeV4VaultProxyAddress, amountIn, amountOut);
+}
+
+  function _swapEnzymeVaultShareForETH(
+    AssetData memory assetData,
+    uint amountIn,
+    uint amountOutMin
```

Comment:
do you see any issues with passing `amountOutMin` in?

It's all controlled by swap operator currently